haystack/core/pipeline/breakpoint.py

```diff
@@ -82,58 +89,26 @@ def _validate_components_against_pipeline(pipeline_snapshot: Dict[str, Any], gra
         )
 
     # Validate 'component_visits'
-    invalid_component_visits = set(pipeline_state["component_visits"].keys()) - valid_components
+    invalid_component_visits = set(pipeline_state.component_visits.keys()) - valid_components
     if invalid_component_visits:
         raise PipelineInvalidPipelineSnapshotError(
             f"Invalid pipeline snapshot: components {invalid_component_visits} in 'component_visits' "
             f"are not part of the current pipeline."
         )
 
+    if isinstance(pipeline_snapshot.break_point, AgentBreakpoint):
+        component_name = pipeline_snapshot.break_point.agent_name
+        visit_count = pipeline_snapshot.break_point.break_point.visit_count
+    else:
+        component_name = pipeline_snapshot.break_point.component_name
+        visit_count = pipeline_snapshot.break_point.visit_count
+
     logger.info(
-        f"Resuming pipeline from component: {pipeline_snapshot['pipeline_breakpoint']['component']} "
-        f"(visit {pipeline_snapshot['pipeline_breakpoint']['visits']})"
+        "Resuming pipeline from {component} with visit count {visits}", component=component_name, visits=visit_count
     )
 
 
-def _validate_pipeline_snapshot(pipeline_snapshot: Dict[str, Any]) -> None:
-    """
-    Validates the loaded pipeline snapshot.
-
-    Ensures that the pipeline_snapshot contains required keys: "input_data", "pipeline_breakpoint",
-    and "pipeline_state".
-
-    Raises:
-        ValueError: If required keys are missing or the component sets are inconsistent.
```

Comment:
This is no longer needed since PipelineSnapshot becomes a dataclass and we get built in checks that all keys are present.